docs/en/latest/plugins/recaptcha.md

```diff
@@ -0,0 +1,116 @@
+| --------- | ------------- | ----------- |----------------------------------------------------------------------------------------------------------------| ------------------------------------------------------------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------- |
+| secret_key | string        | required    |                                                                                                                |  | The secret key of the Google reCAPTCHA v2 service. |
+| parameter_source | string | optional | header                                                                                                         | | The enum of captcha parameter source. Only `header`, `query` are supported. |
+| parameter_name | string | optional | captcha                                                                                                        | | The name of captcha parameter. |
+| response | object | optional    | content_type  = `application/json; charset=utf-8`; status_code = `400`; body = `{"message":"invalid captcha"}` |  | The response for invalid recaptcha token. |
+| ssl_verify | boolean | optional | true                                                                                                           | | verify if SSL cert matches hostname. |
+
+The example configuration of plugin is
+
+```json
+{
+    "secret_key":"6LeIxAcTAAAAAGGXXXXXXXXXXXXXXXXXXX",
+    "parameter_source": "header",
+    "parameter_name": "captcha",
+    "response":{
+        "content_type":"application/json; charset=utf-8",
+        "body":"{\"message\":\"invalid captcha\"}\n",
+        "status_code":400
+    }
+}
+```
+
+## How To Enable
+
+Here's an example, enable this plugin on the specified route:
+
+```shell
+$ curl -i http://127.0.0.1:9080/apisix/admin/routes/1  -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+{
+    "plugins": {
+        "recaptcha": {
+            "secret_key": "6LeIxAcTAAAAAGG-XXXXXXXXXXXXXX",
+            "parameter_source": "header",
+            "parameter_name": "captcha",
+            "response": {
+                "content_type": "application/json; charset=utf-8",
+                "status_code": 400,
+                "body": "{\"message\":\"invalid captcha\"}\n"
+            }
+        }
+    },
+    "upstream": {
+        "nodes": {
+            "127.0.0.1:1980": 1
+        },
+        "type": "roundrobin"
+    },
+    "uri": "/login"
+}'
+```
+
+## Test Plugin
+
+Use curl to request:
+
+```shell
+$ curl -X POST 'http://127.0.0.1:9080/login'
+{"message":"invalid captcha"}
+
+$ curl -X POST 'http://127.0.0.1:9080/login' -H 'captcha: the_invalid_captcha'
+{"message":"invalid captcha"}
```

Comment:
Need a successful case

Reply:
Sorry. It's been a while. I have some personal tasks need to work on.  I just add a successful case, Can you please help to review it?

Reply:
@vm-001 great work 👍